<a href="https://colab.research.google.com/github/Chetana16032002/Chetana16032002/blob/main/internship1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Practise notebook iternship 


In [2]:
import pandas as pd  # here pandas give us the data frame

In [3]:
import numpy as np # numpy is imported becoz in data we got some missing values 

In [4]:
df = pd.DataFrame({
    "Name": ["Hari", "Shyam","Reema", "Bhadur", "Robin", "Arti"],
   "Age": [22, 35, 58, 18, 42, 39],
   "Gender": ["male", "male", "female", "male", "male", "female" ],
    "Income" : [100, 220 , 75, 35, 125, 190], 
    "Car" : [True, np.nan , True, False, True, False]}
   )
df

Name  Age  Gender  Income    Car
0    Hari   22    male     100   True
1   Shyam   35    male     220    NaN
2   Reema   58  female      75   True
3  Bhadur   18    male      35  False
4   Robin   42    male     125   True
5    Arti   39  female     190  False

In [5]:
df.info() #gives the info about missing values ,data types 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6 entries, 0 to 5
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Name    6 non-null      object
 1   Age     6 non-null      int64 
 2   Gender  6 non-null      object
 3   Income  6 non-null      int64 
 4   Car     5 non-null      object
dtypes: int64(2), object(3)
memory usage: 368.0+ bytes


###### non null = not  missing values
##### this implythat in the above table in car there are 5 values out of 6 i.e 1 value is missing 

##### range index = 0-5 i.e 6 rows => it has to be 6 non null other wise we can say that there are some missing values 


*   object=>string or text in pandas 




In [7]:
iris = pd.read_csv(r'https://github.com/YBI-Foundation/Dataset/raw/main/IRIS.csv')

In [9]:
iris.info() # here  iris which is bigger data frame than pandas is been imported 
            # this data set is abut the iris flower 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal_length  150 non-null    float64
 1   sepal_width   150 non-null    float64
 2   petal_length  150 non-null    float64
 3   petal_width   150 non-null    float64
 4   species       150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


*   150 entries => 150 rows 
*   object => string 
*   all rows has 150 non - empty =>no value is missing


In [11]:
df.columns # gives list of names of each column

Index(['Name', 'Age', 'Gender', 'Income', 'Car'], dtype='object')

In [10]:
 df.describe() # gives summary statistics i.e mean , median,mode 

Age      Income
count   6.000000    6.000000
mean   35.666667  124.166667
std    14.486776   69.958321
min    18.000000   35.000000
25%    25.250000   81.250000
50%    37.000000  112.500000
75%    41.250000  173.750000
max    58.000000  220.000000

here there are only 2 columns 

In [13]:
df.describe(include = 'all') #includes all the columns not like above 

Name        Age Gender      Income   Car
count      6   6.000000      6    6.000000     5
unique     6        NaN      2         NaN     2
top     Hari        NaN   male         NaN  True
freq       1        NaN      4         NaN     3
mean     NaN  35.666667    NaN  124.166667   NaN
std      NaN  14.486776    NaN   69.958321   NaN
min      NaN  18.000000    NaN   35.000000   NaN
25%      NaN  25.250000    NaN   81.250000   NaN
50%      NaN  37.000000    NaN  112.500000   NaN
75%      NaN  41.250000    NaN  173.750000   NaN
max      NaN  58.000000    NaN  220.000000   NaN

for a string there cannot be mean ,std ,min.. that is why most of the values in the above table were missing 

here 25,50,75 % are quartiles q1,q2,q3 i.e  q1=>1st quartile ,q2=>median,q3=>3rd
quartile

In [14]:
df

Name  Age  Gender  Income    Car
0    Hari   22    male     100   True
1   Shyam   35    male     220    NaN
2   Reema   58  female      75   True
3  Bhadur   18    male      35  False
4   Robin   42    male     125   True
5    Arti   39  female     190  False

*   we want to extractthe column gender 


In [17]:
df['Gender']

0      male
1      male
2    female
3      male
4      male
5    female
Name: Gender, dtype: object

shape of the data frame => no.o f rows ,columns 


In [18]:
df.shape 


(6, 5)

6,5 => 6 rows and 5 columns 

In [19]:
df['Gender'] # we have to pass information as a list 

0      male
1      male
2    female
3      male
4      male
5    female
Name: Gender, dtype: object

In [20]:
df['Gender'].shape # to derive the shape of the gender column

(6,)

(6,) => 6 rows and 0 columns 
when you are extracting a column using a single bracket you are converting the column vector into a row vector  

->here '.' dot is used to link the function 'shape' '.' acts like a chain

->only df mean that display data frame

->df.shape=>apply shape function on df 


In [23]:
df[['Gender']]

Gender
0    male
1    male
2  female
3    male
4    male
5  female

In [24]:
df[['Gender']].shape

(6, 1)

here becoz of using double bracket we didinot convert column vector to row vector and now the column counts 